CS185Final/app/src/main/java/edu/ucsb/cs/cs185/cs185final/InitialScreen.java

```diff
@@ -0,0 +1,108 @@
+import com.google.android.gms.maps.model.MarkerOptions;
+
+
+public class InitialScreen extends AppCompatActivity {
+    private GoogleMap mMap; // Might be null if Google Play services APK is not available.
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_initial_screen);
+
+        Button newGame = (Button) findViewById(R.id.button);
+        Button joinGame = (Button) findViewById(R.id.button2);
+
+        newGame.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent createGame = new Intent(getApplicationContext(), createGame.class);
+                startActivity(createGame);
+            }
+        });
+
+        joinGame.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent map = new Intent(getApplicationContext(), MapsActivity.class);
+                startActivity(map);
+            }
+        });
+        setUpMapIfNeeded();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_initial_screen, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        setUpMapIfNeeded();
+    }
+
+    void setUpMapIfNeeded() {
+        // Do a null check to confirm that we have not already instantiated the map.
+        if (mMap == null) {
+            // Try to obtain the map from the SupportMapFragment.
+            mMap = ((SupportMapFragment) getSupportFragmentManager().findFragmentById(R.id.map))
+                    .getMap();
+            // Check if we were successful in obtaining the map.
+            if (mMap != null) {
+                setUpMap();
+            }
+        }
+    }
+
+    private void setUpMap() {
+        LocationManager locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+        Criteria criteria = new Criteria();
+        String locationProvider = locationManager.getBestProvider(criteria, false);
+        Location location = locationManager.getLastKnownLocation(locationProvider);
+        LatLng latlng = new LatLng(location.getLatitude(), location.getLongitude());
```

Comment:
can we set default location everywhere, otherwise it crashes on my phone:
it will be something like this,

LatLng latlng;
        if(location != null){
            latlng = new LatLng(location.getLatitude(), location.getLongitude());
        }
        else latlng = new LatLng(0,0);